src/msg/async/dpdk/dpdk_rte.cc

```diff
@@ -122,12 +123,14 @@ namespace dpdk {
           funcs.pop_front();
           f();
           cond.notify_all();
+          if (exit_thread) {
```

Comment:
i think it's just weird to post a "exit" func to the worker thread, and check this flag after calling `f()`. instead, i recommend:
```c++
for (;;) {
  cond.wait(l, [this]{ return !funcs.empty() || stopped; });
  if (!funcs.empty()) {
    // ...
  } else {
    assert(stopped);
    break;
  }
}
```

as we cannot rely on funcs to trigger the check.

Reply:
good suggestion